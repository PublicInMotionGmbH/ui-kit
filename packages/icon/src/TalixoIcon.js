import React from 'react'
import PropTypes from 'prop-types'

import { buildClassName } from '@talixo/shared'
import icons from '@talixo/icon-pack'

const iconPrefix = icons.classPrefix
const iconsMap = icons.icons

const propTypes = {
  /** Icon name */
  name: PropTypes.string.isRequired,

  /** Additional class name */
  className: PropTypes.string
}

/**
 * Component which represents icon from Talixo set
 *
 * @param {object} props
 * @param {string} [props.name]
 * @param {string} [props.className]
 * @returns {React.Element}
 */
function TalixoIcon (props) {
  const { className, name, ...passedProps } = props

  const clsName = buildClassName('icon', [ className, iconPrefix ])

  return (
    <span className={clsName} {...passedProps} aria-hidden='true'>{iconsMap[name]}</span>
  )
}

TalixoIcon.propTypes = propTypes

export default TalixoIcon
