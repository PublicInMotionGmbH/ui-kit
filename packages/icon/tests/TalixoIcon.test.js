import React from 'react'
import { shallow } from 'enzyme'

import TalixoIcon from '../src/TalixoIcon'

describe('<TalixoIcon />', () => {
  it('renders children correctly', () => {
    const wrapper = shallow(<TalixoIcon name='affiliate' />)

    expect(wrapper).toMatchSnapshot()
  })
})
