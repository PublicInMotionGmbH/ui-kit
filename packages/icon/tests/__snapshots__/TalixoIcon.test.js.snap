// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders Material icon correctly 1`] = `
<span
  aria-hidden="true"
  className="talixo-icon tai tai-affiliate"
/>
`;
